mk/spksrc.wheel.mk

```diff
@@ -76,7 +76,7 @@ build_wheel_target: $(PRE_WHEEL_TARGET)
 			if [ -z "$(CROSSENV)" ]; then \
 				$(RUN) _PYTHON_HOST_PLATFORM="$(TC_TARGET)" CFLAGS="$(CFLAGS) -I$(STAGING_INSTALL_PREFIX)/$(PYTHON_INC_DIR) $(WHEELS_CFLAGS)" LDFLAGS="$(LDFLAGS) $(WHEELS_LDFLAGS)" $(PIP_WHEEL) --requirement $(WHEELHOUSE)/$(WHEELS_CROSS_COMPILE) ; \
 			else \
-				. $(CROSSENV) && $(RUN) _PYTHON_HOST_PLATFORM="$(TC_TARGET)" CFLAGS="$(CFLAGS) -I$(STAGING_INSTALL_PREFIX)/$(PYTHON_INC_DIR) $(WHEELS_CFLAGS)" LDFLAGS="$(LDFLAGS) $(WHEELS_LDFLAGS)" $(PIP_WHEEL) --no-build-isolation --requirement $(WHEELHOUSE)/$(WHEELS_CROSS_COMPILE) ; \
+				. $(CROSSENV) && $(RUN) _PYTHON_HOST_PLATFORM="$(TC_TARGET)" CFLAGS="$(CFLAGS) -I$(STAGING_INSTALL_PREFIX)/$(PYTHON_INC_DIR) $(WHEELS_CFLAGS)" LDFLAGS="$(LDFLAGS) $(WHEELS_LDFLAGS)" pip $(PIP_WHEEL_ARGS) --no-build-isolation --requirement $(WHEELHOUSE)/$(WHEELS_CROSS_COMPILE) ; \
```

Comment:
@th0ma7 Ah found it! `pip` and `$(PIP)` are not the same. Or at least `pip` works and `$(PIP)` doesn't

Reply:
Are we in a virtual environment?

Reply:
Sorry @th0ma7 I'm going to revert the 2 files (mk/spksrc.common.mk,  mk/spksrc.wheel.mk) from https://github.com/SynoCommunity/spksrc/commit/47cb2579fb5a7c543d4c94b3161c1f696510c48c#diff-b2c976fb01fa035ca37be782346b9773732c78d931102770d1a305d9cfdc6a06

Reply:
I'll have to go over the various changes you did (any left or not?) so my other PR doesn't get too funky neither.
I'm glad you have sorted it out, good job!

Reply:
Re-reading it I don't understand:
1. why would it work using `pip` instead of `$(PIP)` when really it's the same thing?  If not we need to find out what's the difference and why (and probably add a comment on top of `PIP ?= pip` to explain it)
2. Otherwise my "guess" is that the re-ordering in `PIP_WHEEL_ARGS` is what fixed the issue.  On the other hand, if it was only `pip` then why changing that?
3. Currently `PIP_DONWLOAD` calls `PIP_ARGS` which is now existent (probably easy to fix)

Reply:
Before I rebase against master my PR, can you double-check this and confirm a few things?  Otherwise I fear I may break things back for `sickchill` while fixing `PIP_DOWNLOAD`?

Reply:
Wow, just wow!
@publicarray I've been double-checking and I just don't get it... it does solves the issue no matter why?

I've checked the following:
```
                which $(PIP) ;\
                which pip ;\
                $(MSG) $(PIP_WHEEL) ;\
                $(MSG) $(PIP) $(PIP_WHEEL_ARGS) ;\
                $(MSG) pip $(PIP_WHEEL_ARGS) ;\
```

Which gives this:
```
/usr/local/bin/pip
/usr/local/bin/pip
===>  pip wheel --no-binary :all: --cache-dir /home/spksrc/python310-fixes/spksrc/spk/sickchill/../../distrib/pip --no-deps --wheel-dir /home/spksrc/python310-fixes/spksrc/spk/sickchill/work-x64-7.0/wheelhouse
===>  pip wheel --no-binary :all: --cache-dir /home/spksrc/python310-fixes/spksrc/spk/sickchill/../../distrib/pip --no-deps --wheel-dir /home/spksrc/python310-fixes/spksrc/spk/sickchill/work-x64-7.0/wheelhouse
===>  pip wheel --no-binary :all: --cache-dir /home/spksrc/python310-fixes/spksrc/spk/sickchill/../../distrib/pip --no-deps --wheel-dir /home/spksrc/python310-fixes/spksrc/spk/sickchill/work-x64-7.0/wheelhouse
```

Reply:
Found it!!!!! :smile: 
- The defined `$(PIP)` in `spksrc.common.mk` points to the native binary.
- When direct `pip` is called with `. $(CROSSENV) && $(RUN) ...` it points to cross-environment as the path gets changed such as:
```
$ cat work-x64-7.0/crossenv/bin/activate
. /home/spksrc/python310-fixes/spksrc/spk/sickchill/work-x64-7.0/crossenv/cross/bin/activate
export PATH=/home/spksrc/python310-fixes/spksrc/spk/sickchill/work-x64-7.0/crossenv/bin:$PATH
```

Resulting in:
```
::::::::::::::
work-x64-7.0/pip.1
::::::::::::::
pip 21.3 from /home/spksrc/python310-fixes/spksrc/native/python38/work-native/install/usr/local/lib/python3.8/site-packages/pip (python 3.8)
::::::::::::::
work-x64-7.0/pip.2
::::::::::::::
pip 21.3 from /home/spksrc/python310-fixes/spksrc/spk/sickchill/work-x64-7.0/crossenv/build/lib/python3.8/site-packages/pip (python 3.8)
```